modules/questions.py

```diff
@@ -32,226 +27,418 @@
+                cast(dict, question_row)
+            )
+            why = dedent(
+                f"""{message.author} asked me to set the status of question
+                with id "{query.id}" to "{query.status}" and so I did"""
+            )
+        return Response(confidence=8, text=response_msg, why=why)
+
+    def update_question_status(
+        self, query: SetQuestionQuery, message: ServiceMessage
+    ) -> str:
+        row_id = query.id
+        uri = f"https://coda.io/apis/v1/docs/{self.DOC}/tables/{self.ALL_ANSWERS_TABLE}/rows/{row_id}"
+        payload = {
+            "row": {
+                "cells": [
+                    {"column": "Status", "value": query.status},
+                ],
+            },
+        }
+        req = requests.put(uri, headers=self.get_headers(), json=payload)
+        if query.type == "@":
+            msg = f"Thanks, @{message.author.name}!\nI updated this question's status to `{query.status}`"
+        else:
+            msg = f"Updated question's  status to `{query.status}`"
+        log.info(msg)
+        return msg
 
     ############
     # Coda API #
     ############
 
     @classmethod
-    def send_coda_questions_request(cls, qq: QuestionQuery | None = None) -> dict:
+    def send_all_answers_table_request(
+        cls, query: Optional[PostOrCountQuestionQuery] = None
+    ) -> dict:
         """Get rows from "All Answers" table in our coda"""
-
-        headers = {"Authorization": f"Bearer {cls.CODA_API_TOKEN}"}
         params = {
             "valueFormat": "simple",
             "useColumnNames": True,
             "visibleOnly": False,
             "limit": 1000,
         }
         # optionally query by status
-        if qq and qq.status:
-            params["query"] = f'"Status":"{qq.status}"'
-        uri = f"https://coda.io/apis/v1/docs/{cls.DOC_ID}/tables/{cls.TABLE_ID}/rows"
-        response = requests.get(uri, headers=headers, params=params, timeout=16).json()
+        if query and query.status:
+            params["query"] = f'"Status":"{query.status}"'
+        uri = f"https://coda.io/apis/v1/docs/{cls.DOC}/tables/{cls.ALL_ANSWERS_TABLE}/rows"
+        response = requests.get(
+            uri, headers=cls.get_headers(), params=params, timeout=16
+        ).json()
         return response
 
+    @classmethod
+    def get_questions_df(
+        cls, query: Optional[PostOrCountQuestionQuery] = None
+    ) -> pd.DataFrame:
+        """Get questions from with `status="Not started"`"""
+        request_res = cls.send_all_answers_table_request(query)
+        rows = [parse_row(row) for row in request_res["items"]]
+        return pd.DataFrame(rows)
+
+    def get_question_by_id(self, row_id: str) -> Optional[ParsedRow]:
+        """Get question by id in "All Answers" table.
+        Returns `ParsedRow` or `None` (if question with that id doesn't exist)
+        """
+        uri = f"https://coda.io/apis/v1/docs/{self.DOC}/tables/{self.ALL_ANSWERS_TABLE}/rows/{row_id}"
```

Comment:
hm, we have so many lines with variations of Coda URLs ... do we want to extract these to some Coda API utility file? maybe in a future issue and maybe @mruwnik can help with suggestions to keep chatbot and UI code consistent how we communicate with Coda?